frontend/src/lib/components/UserSettings.svelte

```diff
@@ -0,0 +1,156 @@
+<svelte:options runes={true} />
+
+<script lang="ts">
+import {
+	authCookieLogin,
+	usersCurrentUser,
+	usersPatchCurrentUser,
+} from "$lib/client/services.gen";
+import { i18n } from "$lib/i18n.svelte";
+import { preventDefault } from "$lib/util";
+import { Button, Heading, Input, Modal } from "flowbite-svelte";
+import AlertMessage from "./AlertMessage.svelte";
+
+let newPassword = $state(null) as string | null;
+let newPasswordRepeat = $state(null) as string | null;
+let currentPassword = $state(null) as string | null;
+let currentPasswordValid = $state(false);
+let passwordChangeSuccess = $state(false);
+let showAlert = $state(false);
+let alertMessage = $state(i18n.tr.settings.defaultAlertMessage);
+
+async function submitNewPassword() {
+	currentPasswordValid = false;
+	passwordChangeSuccess = false;
+	showAlert = false;
+
+	if (currentPassword === null || currentPassword === "") {
+		showAlert = true;
+		alertMessage = i18n.tr.settings.emptyPasswordError;
+		return;
+	}
+
+	if (newPassword === null || newPassword === "") {
+		showAlert = true;
+		alertMessage = i18n.tr.settings.emptyPasswordError;
+		return;
+	}
+
+	if (newPassword !== newPasswordRepeat) {
+		showAlert = true;
+		alertMessage = i18n.tr.settings.nonMatchingPasswordsError;
+		return;
+	}
+
+	if (newPassword === currentPassword) {
+		showAlert = true;
+		alertMessage = i18n.tr.settings.samePasswordsError;
+		return;
+	}
+
+	const currentUser = await usersCurrentUser();
+
+	if (currentUser.error || !currentUser.data) {
+		alertMessage = i18n.tr.settings.getUserError;
+		showAlert = true;
+		return;
+	}
+
+	const verifyResponse = await authCookieLogin({
+		body: {
+			username: currentUser.data.email,
+			password: currentPassword,
+		},
+	});
+
+	if (verifyResponse.error) {
+		alertMessage = i18n.tr.settings.oldPasswordWrong;
+		showAlert = true;
+		return;
+	}
+
+	currentPasswordValid = true;
+
+	const patchResponse = await usersPatchCurrentUser({
+		body: {
+			password: newPassword,
+		},
+	});
+
+	if (patchResponse.error) {
+		alertMessage = i18n.tr.settings.sendError;
+		showAlert = true;
+		passwordChangeSuccess = false;
+	} else {
+		passwordChangeSuccess = true;
+	}
+}
+</script>
+
+{#if showAlert}
+	<AlertMessage
+		id="alertMessageSettings"
+		title={i18n.tr.settings.alertTitle}
+		message={alertMessage}
+		onclick={() => {
+			showAlert = false;
+		}}
+	/>
+{/if}
+
+<div class="m-2 p-2 flex flex-col space-y-2 text-gray-700 dark:text-gray-400">
+	<Heading tag="h4" class="font-bold text-gray-700 dark:text-gray-400" id="changePasswordHeading"
+		>{i18n.tr.settings.changePassword}</Heading
+	>
+	<form
+		class="space-y-4 mb-2 pb-2"
+		onsubmit={preventDefault(submitNewPassword)}
+	>
+		<Input
```

Comment:
@MaHaWo I thought I already wrote this review comment but apparently it got lost somehow, maybe when my pc restarted accidentally...

Could you add labels and autocompletes as we do in the user registration form: 
https://github.com/ssciwr/mondey/blob/dc96a36f8788b7a9a1041b78707ba54b0e0d1d06/frontend/src/lib/components/UserRegistration.svelte#L69-L83

Otherwise lgtm, thanks!